addons/mail/models/mail_activity.py

```diff
@@ -166,6 +171,11 @@ def _default_activity_type_id(self):
         'res.users', 'Assigned to',
         default=lambda self: self.env.user,
         index=True, required=True)
+    tz = fields.Selection(
```

Comment:
If you really want to go that way (which I am unsure of), you should make a specific commit to handle tz in activities. Notably it should be aligned with assigned user and not with current user (could be admin or root in automated processes). You should also probably think with functional experts what to do with "UserA in TZA encodes for UserB in TZB".